### PR TITLE
Fixing connection without saved password not connecting in Async Server Tree

### PR DESCRIPTION
**Pull request by aasimkhan30**
1. Bug 1: fixing connections with forgot password not connecting.

Previously, we encountered a bug when attempting to connect a profile that didn't have a saved password. Our usual workflow involved opening a connection dialog and prompting users to enter their password before connecting to the server. In such cases, the 'onConnectionProfileConnected' event was triggered. However, this created an issue since the connection returned from the connection dialog was completely new, and we couldn't locate it in the tree.

This PR modifies the connection management service. Instead of triggering the 'onConnectionProfileConnected' event, the service now triggers an 'onConnectionProfileEdited' event. This event includes the profile ID of the old connection in the tree and the new connection from the connection dialog. The 'onConnectionProfileEdited' handler is then able to locate the old connection in the tree and replace it with the new connection from the dialog.

This should not interfere with the old tree as the old tree is not using 'onConnectionProfileConnected' and 'onConnectionProfileEdited' events. 

2. Bug 2:
When disconnecting connections where the 'save password' option is set to false, we now remove the saved password from the connection.

### CODE REVIEW

File: src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts:475]: What was the bug you were seeing with this?

Comment (aasimkhan30) [reply src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts:475]: ads was not asking for the password the next time the user connected with this connection. Since the password was already saved in the connection from last session.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts:475]: This is only an issue in the async tree I'm guessing then, right? Because the other tree makes new inputs when things change?

Comment (aasimkhan30) [reply src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts:475]: Yup, an async tree only issue since we do not get rid of connections objects when they are disconnected.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts:475]: Could you add that as a comment here so we aren't confused later on why this is happening?

**Review by Charles-Gagnon — APPROVED** [review]
Although I think it'd be worth it to wait until code complete cutoff to merge any more changes to OE stuff - the risk of unintended regressions is pretty high....

**Pull request merged**
